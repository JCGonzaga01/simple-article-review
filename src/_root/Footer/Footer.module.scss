@import "helpers/stylings/mixins";
@import "helpers/stylings/global";

.wrapper {
  @include align-center;
  border-top: 2px solid $black;
  justify-content: space-between;
  margin-bottom: 50px;
  padding-top: 25px;

  .adrenalin {
    height: 14px;
    width: 83px;
  }

  .footerLinks {
    @include media-query(sp) {
      display: flex;
      flex-direction: column;
    }

    color: $black;
    font-family: $fontType1;
    font-size: $fontXS1;
    font-weight: 400;
    line-height: $lineheightXS1;

    & > span {
      @include media-query(sp) {
        margin-right: 0;
        margin-bottom: 10px;
      }

      &:hover {
        cursor: pointer;
        background: $black;
        color: $white;
        border-radius: 3px;
      }

      &:not(:last-child) {
        margin-right: 35px;
      }
    }
  }
}
