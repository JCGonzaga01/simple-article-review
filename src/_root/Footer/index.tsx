import React from "react";
import { footerLinks } from "constants/global";
import styles from "./Footer.module.scss";

const Footer: React.FC = () => (
  <div className={styles.wrapper}>
    <img
      className={styles.adrenalin}
      src={process.env.PUBLIC_URL + "/assets/adrenalin.svg"}
      alt={"adrenalin"}
    />
    <div className={styles.footerLinks}>
      {footerLinks.map((text, idx) => (
        <span key={idx}>{text}</span>
      ))}
    </div>
  </div>
);

export default Footer;
