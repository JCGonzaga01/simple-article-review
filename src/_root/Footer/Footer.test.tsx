import React from "react";
import Footer from "./index";
import renderer from "react-test-renderer";

describe("<Footer />", () => {
  let wrapper: any;
  let instance: any;

  beforeEach(() => {
    wrapper = renderer.create(<Footer />);
    instance = wrapper.root;
  });

  test("should mount component", () => {
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should display footer links", () => {
    expect(instance.findByProps({ className: "footerLinks" }).children).toHaveLength(6);
  });
});
