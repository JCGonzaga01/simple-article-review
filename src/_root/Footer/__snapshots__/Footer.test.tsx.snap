// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should mount component 1`] = `
<div
  className="wrapper"
>
  <img
    alt="adrenalin"
    className="adrenalin"
    src="/assets/adrenalin.svg"
  />
  <div
    className="footerLinks"
  >
    <span>
      Privacy
    </span>
    <span>
      Sitemap
    </span>
    <span>
      Facebook
    </span>
    <span>
      LinkedIn
    </span>
    <span>
      Instagram
    </span>
    <span>
      Twitter
    </span>
  </div>
</div>
`;
